core/src/main/java/com/linecorp/armeria/server/HttpResponseException.java

```diff
@@ -19,12 +19,40 @@
 
 import com.linecorp.armeria.common.Flags;
 import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.util.Exceptions;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
 
 /**
  * A {@link RuntimeException} that is raised when an armeria internal http exception has occurred.
  * This class is the general class of exceptions produced by a failed request or a reset stream.
  */
 public class HttpResponseException extends RuntimeException {
+
+    private static final Int2ObjectMap<HttpResponseException> EXCEPTIONS = new Int2ObjectOpenHashMap<>();
+
+    /**
+     * Returns a new {@link HttpResponseException} instance with the HTTP status code.
+     */
+    public static HttpResponseException of(int statusCode) {
+        return of(HttpStatus.valueOf(statusCode));
+    }
+
+    /**
+     * Returns a new {@link HttpResponseException} instance with the {@link HttpStatus}.
+     */
+    public static HttpResponseException of(HttpStatus httpStatus) {
+        requireNonNull(httpStatus, "httpStatus");
+        if (Flags.verboseExceptions()) {
+            return new HttpResponseException(httpStatus);
+        } else {
+            final int statusCode = httpStatus.code();
+            return EXCEPTIONS.computeIfAbsent(statusCode, code ->
+                    Exceptions.clearTrace(new HttpResponseException(code)));
```

Comment:
Oops.. I had to check its thread safety first. :-( I changed the map implementation to `ConcurrentHashMap`.